src/sql/src/func.rs

```diff
@@ -1664,7 +1664,21 @@ lazy_static! {
 
             // Aggregates.
             "array_agg" => Aggregate {
-                params!(Any) => Operation::unary(|_ecx, _e| unsupported!("array_agg")), 4053;
+                params!(NonVecAny) => Operation::unary(|ecx, e| {
+                    // TODO(#7572): remove this
+                    let e = match ecx.scalar_type(&e) {
+                        ScalarType::Char { length } => e.call_unary(UnaryFunc::PadChar { length }),
+                        _ => e,
+                    };
+                    // ArrayAgg excepts all inputs to be arrays, so wrap all input datums into
+                    // arrays.
+                    let e_arr = HirScalarExpr::CallVariadic{
+                        func: VariadicFunc::ArrayCreate{elem_type: ecx.scalar_type(&e)},
+                        exprs: vec![e],
+                    };
+                    Ok((e_arr, AggregateFunc::ArrayAgg))
+                }), 2335;
+                params!(ArrayAny) => Operation::unary(|_ecx, _e| unsupported!("array_agg")), 4053;
```

Comment:
Can we remove this impl to make the error message clearer, i.e. that we don't support array_agg on arrays? Or expand the text of the call to `unsupported` to add more color?

Reply:
Added text to mention arrays are unsupported.